sycl/include/sycl/__spirv/spirv_ops.hpp

```diff
@@ -238,69 +238,53 @@ extern __DPCPP_SYCL_EXTERNAL
   extern __DPCPP_SYCL_EXTERNAL Type __spirv_AtomicSMin(                        \
-      AS Type *P, __spv::Scope::Flag S, __spv::MemorySemanticsMask::Flag O,    \
-      Type V);
+      AS Type *P, int S, int O, Type V) noexcept;
 #define __SPIRV_ATOMIC_UMIN(AS, Type)                                          \
   extern __DPCPP_SYCL_EXTERNAL Type __spirv_AtomicUMin(                        \
-      AS Type *P, __spv::Scope::Flag S, __spv::MemorySemanticsMask::Flag O,    \
-      Type V);
+      AS Type *P, int S, int O, Type V) noexcept;
 #define __SPIRV_ATOMIC_FMIN(AS, Type)                                          \
   extern __DPCPP_SYCL_EXTERNAL Type __spirv_AtomicFMinEXT(                     \
-      AS Type *P, __spv::Scope::Flag S, __spv::MemorySemanticsMask::Flag O,    \
-      Type V);
+      AS Type *P, int S, int O, Type V) noexcept;
 #define __SPIRV_ATOMIC_SMAX(AS, Type)                                          \
   extern __DPCPP_SYCL_EXTERNAL Type __spirv_AtomicSMax(                        \
-      AS Type *P, __spv::Scope::Flag S, __spv::MemorySemanticsMask::Flag O,    \
-      Type V);
+      AS Type *P, int S, int O, Type V) noexcept;
 #define __SPIRV_ATOMIC_UMAX(AS, Type)                                          \
   extern __DPCPP_SYCL_EXTERNAL Type __spirv_AtomicUMax(                        \
-      AS Type *P, __spv::Scope::Flag S, __spv::MemorySemanticsMask::Flag O,    \
-      Type V);
+      AS Type *P, int S, int O, Type V) noexcept;
 #define __SPIRV_ATOMIC_FMAX(AS, Type)                                          \
   extern __DPCPP_SYCL_EXTERNAL Type __spirv_AtomicFMaxEXT(                     \
-      AS Type *P, __spv::Scope::Flag S, __spv::MemorySemanticsMask::Flag O,    \
-      Type V);
+      AS Type *P, int S, int O, Type V) noexcept;
 #define __SPIRV_ATOMIC_AND(AS, Type)                                           \
-  extern __DPCPP_SYCL_EXTERNAL Type __spirv_AtomicAnd(                         \
-      AS Type *P, __spv::Scope::Flag S, __spv::MemorySemanticsMask::Flag O,    \
-      Type V);
+  extern __DPCPP_SYCL_EXTERNAL Type __spirv_AtomicAnd(AS Type *P, int S,       \
+                                                      int O, Type V) noexcept;
 #define __SPIRV_ATOMIC_OR(AS, Type)                                            \
-  extern __DPCPP_SYCL_EXTERNAL Type __spirv_AtomicOr(                          \
-      AS Type *P, __spv::Scope::Flag S, __spv::MemorySemanticsMask::Flag O,    \
-      Type V);
+  extern __DPCPP_SYCL_EXTERNAL Type __spirv_AtomicOr(AS Type *P, int S, int O, \
+                                                     Type V) noexcept;
 #define __SPIRV_ATOMIC_XOR(AS, Type)                                           \
-  extern __DPCPP_SYCL_EXTERNAL Type __spirv_AtomicXor(                         \
-      AS Type *P, __spv::Scope::Flag S, __spv::MemorySemanticsMask::Flag O,    \
-      Type V);
+  extern __DPCPP_SYCL_EXTERNAL Type __spirv_AtomicXor(AS Type *P, int S,       \
```

Comment:
For my education, why do we still need these?

Reply:
@aelovikov-intel it is related to long long type, please check comment: https://github.com/intel/llvm/pull/17471#discussion_r2023862803